src/smirnoffio.py

```diff
@@ -433,13 +495,33 @@ def update_simulation(self, **kwargs):
         #         delattr(self, 'simulation')
         # self.vsprm = vsprm.copy()
 
+        if openff_topology.n_topology_virtual_sites > 0:
+            # For now always assume that the v-sites have changed. This is currently
+            # needed as the FB checks don't support the ``LocalCoordinatesSite`` based
+            # virtual sites that OpenFF uses.
+            if hasattr(self, 'simulation'):
+                delattr(self, 'simulation')
+
```

Comment:
I'm wondering why you had to do that? Is there no way to update the virtual site parameters without deleting the Simulation object? In practice does it affect performance?

Reply:
I believe I did this as the `PrepareVirtualSites` function did not yet support `LocalCoordinatesSite` which the OpenFF toolkit seems to use exclusively. Once #215 is merged however it would seem that this might not be needed.